### PR TITLE
Initialise react routes and create layout folder

### PR DESCRIPTION
**Pull request by AhmedQeshta**
## Initialise react routes 

 - install package [react-outer](https://reactrouter.com/)
 - create Layout Folder `src/layouts`
 - add to layout files
     -  client Layout (start up pages - [logins- filter page])
     -  dashboard Layout

> When user go to `/` or `/login` or `/gym/login` and so on, must render component inside client layout by `<Outlet/>`,
and if the user got to `/dashboard/*` must render Component in side Dashboard Layout using `<Outlet/>`.

and the last thing I Create two functions named `InnerComponent`, `NotFound` inside the file `App.js`, this is for the test is Outlet is working correctly this Function must be pages a user goes to it.

### CODE REVIEW

File: client/.eslintrc.json [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+    "ecmaVersion": "latest",
+    "sourceType": "module"
+  },
+  "plugins": ["react", "prettier", "jsx-a11y"],
+  "rules": {
+    "no-console": "warn",
+    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
+    "prettier/prettier": [
+      "error",
+      {
```
Comment (LinaYahya) [inline client/.eslintrc.json:27]: Not sure why do we need prettier!

File: client/src/App.js [hunk 2]
```diff
@@ -1,13 +1,36 @@
 import React from 'react';
+import { Routes, Route } from 'react-router-dom';
 import './app.css';
+import { Dashboard, Home } from './layouts';
+import { LoginPage, RegisterPage, LoginGymPage, GymPage, NotfoundPage } from './pages';
```
Comment (LinaYahya) [inline client/src/App.js:5]:
```suggestion
import { LoginPage, RegisterPage, LoginGymPage, GymPage, NotfoundPage } from './Pages';
```
Linux is a case-sensitive in files naming, So take care about this

File: client/src/layouts/Dashboard.jsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Outlet } from 'react-router-dom';
+
+function Dashboard() {
+  return (
+    <div className="container">
```
Comment (LinaYahya) [inline client/src/layouts/Dashboard.jsx:6]: as you changed the class name change it in the app.css also

File: client/src/App.js [hunk 4]
```diff
@@ -1,13 +1,36 @@
...
-      <h1>Welcome To FitHouse</h1>
-      <h2> Hi</h2>
-    </div>
+    <Routes>
+      {/* Routes For Site Views {login, gym filter,...} */}
+      <Route path="/" element={<Home />}>
+        <Route index element={<LoginPage />} />
+        <Route path="login" element={<LoginPage />} />
+        <Route path="register" element={<RegisterPage />} />
+        <Route path="gym/login" element={<LoginGymPage />} />
```
Comment (LinaYahya) [inline client/src/App.js:15]: we still have a search route and gym profile
Add them, and give all your pages a container className

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/.eslintrc.json (hunk 1 above)
Comment (AhmedQeshta) [reply client/.eslintrc.json:27]: this is for `import` single `'` and other double `"`. 
but is not important, I will remove it.

File: client/src/layouts/Dashboard.jsx (hunk 3 above)
Comment (AhmedQeshta) [reply client/src/layouts/Dashboard.jsx:6]: okay, I will do it now. 👨🏻‍💻

**Review by AFahgan — APPROVED** [review]

**Review by rawaneyad — APPROVED** [review]
Well done

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**